### DISCUSSION TITLE
Web Socket is closed the connection was closed abnormally (Error 1006) | Invalid parse status: 400

### DISCUSSION THREAD

**Discussion by PanterSoft** [OvenMediaLabs/OvenMediaEngine#1183] [Help] [answered]
Hello,

I am trying to stream my webcam via the WebRTC Input (Beta) but I get the Error 

>  Web Socket is closed. The connection was closed abnormally, e.g., without sending or receiving a Close control frame

When I stream to the OvenPlayer Demo everything works fine but when I try with my own instance then I get the error.

Here is my Ovenmediaengine Logfile:
[ovenmediaengine.log](https://github.com/AirenSoft/OvenMediaEngine/files/11198594/ovenmediaengine.log)

Here is my Server Config file:
[Server.xml.txt](https://github.com/AirenSoft/OvenMediaEngine/files/11198597/Server.xml.txt)

**Comment by getroot** [comment]
Your secure websocket port is 3334. But you might try to connect 3333 port with wss. try to connect wss://host:3334/app/stream

**Reply by PanterSoft** [reply]
Hi, thanks for the quick reply.

yeah you are right I tried it before and it was also not working. but now I got a different result for which ever reason. now my problem is that I am getting the error 

> 

> Websocket is Closed. The connection was closed due to a failure to perform a TLS handshake (e.g. the server certificate can't be verified)

For context my TLS certificate is generated following this Tutorial:
[openssl-create-certificate](https://www.golinuxcloud.com/openssl-create-certificate-chain-linux/)

**Reply by naanlizard** [reply]
Are you trying to connect over wss to the ws (insecure) port?

**Reply by getroot** [reply]
If you are using a private certificate, your browser will reject it.

**Reply by PanterSoft** [reply, marked as answer]
@naanlizard In the "Ovenplayer Demo Input Beta" only wss is allowed so I connect via that. The port which I am using is 3334

@getroot okay, actually I don't know exactly If I have a private certificate but I will check that

I found that in the latest Logfile:

> [2023-04-11 13:21:42.359] E [SPRtcSig-t3334:10657] OpenSSL | tls.cpp:201  | An error occurred while accept SSL connection: [OpenSSL] error:0A000412:SSL routines::sslv3 alert bad certificate (167773202)

Is this error related with a private certificate ? If yes how can I create a public certificate which will be accepted by my browser ? can you give me a small push in the right direction ?

Edit:

I got it fixed. I just have used the wrong certificate. after using my lets encrypt certificate it works fine

thanks for your help and time
